pymeasure/instruments/exfo/ctp10.py

```diff
@@ -0,0 +1,1031 @@
+
+        Note: This command does not apply to VIAVI mSWS-AISLS lasers.
+
+        :return: Current trigger input setting (int, 0-8).
+
+        Example:
+            # Set to no trigger
+            ctp.tls[1].trigin = 0
+
+            # Set to use TRIG IN port 3
+            ctp.tls[1].trigin = 3
+
+            # Query current trigger setting
+            trig = ctp.tls[1].trigin  # returns 0-8
+        """,
+        validator=strict_range,
+        values=[0, 8],
+        cast=int,
+    )
+
+
+class DetectorChannel(Channel):
+    """Represents a detector channel on the EXFO CTP10.
+
+    A detector channel corresponds to a physical detector on a module and provides
+    access to both detector-level operations (power, trigger, units, references) and
+    trace data retrieval for specific trace types.
+
+    The channel is identified by:
+    - SENSe[1...20]: Module identification number (position in mainframe, left to right)
+        * In Daisy chaining mode: positions 1-10 are Primary mainframe, 11-20 are Secondary
+    - CHANnel[1...6]: Detector identification number (detector position on module, top to bottom)
+
+    Trace types (TYPE parameter):
+    - TYPE1: TF live trace (Transmission Function)
+    - TYPE11: Raw Live trace
+    - TYPE12: Raw Reference trace
+    - TYPE13: Raw Quick Reference trace
+    - See instrument documentation for complete list (1-23)
+
+    Example:
+        # Access detector on module 4, channel 1
+        detector = ctp.detector(module=4, channel=1)
+
+        # Detector-level operations
+        detector.power_unit = 'DBM'  # Set unit to dBm
+        power_dbm = detector.power  # Get optical power
+        detector.trigger = 0  # Set software trigger
+        detector.create_reference()  # Create reference
+
+        # Read trace data for TF live trace (TYPE1)
+        length = detector.length(trace_type=1)
+        wavelengths = detector.get_data_x(trace_type=1, unit='M', format='BIN')
+        powers = detector.get_data_y(trace_type=1, unit='DB', format='BIN')
+
+        # Read Raw Live trace (TYPE11)
+        raw_powers = detector.get_data_y(trace_type=11, unit='DB', format='BIN')
+
+        # Save trace data
+        detector.save(trace_type=1)
+    """
+
+    def __init__(self, parent, id=None, **kwargs):
+        """Initialize detector channel with module and channel.
+
+        :param tuple id: Tuple of (module, channel) or None for dynamic access.
+            module (int): SENSe number 1-20 (module position in mainframe).
+            channel (int): CHANnel number 1-6 (detector position on module).
+        """
+        # Accept tuple or None (for ChannelCreator pattern)
+        if id is not None and not (isinstance(id, tuple) and len(id) == 2):
+            raise ValueError("DetectorChannel ID must be a tuple of (module, channel)")
+
+        # Store module and channel as separate attributes
+        if id is not None:
+            self.module, self.channel = id
```

Comment:
Why do you need to store it? The id will be stored in self.id during the super call.